libs/state/src/lib/provide-rx-state-config.ts

```diff
@@ -0,0 +1,98 @@
+import { InjectionToken, Provider } from '@angular/core';
+import {
+  AccumulationFn,
+  defaultAccumulator,
+} from '@rx-angular/state/selections';
+import { queueScheduler, SchedulerLike } from 'rxjs';
+
+export const enum RX_STATE_CONFIGS {
+  Scheduler,
+  Accumulator,
+}
+
+interface RxStateConfigFn {
+  kind: RX_STATE_CONFIGS;
+  providers: Provider[];
+}
+
+/**
+ * Injection token for the default accumulator function.
+ *
+ * @example
+ * providers: [
+ *  {
+ *   provide: RX_ACCUMULATOR_FN,
+ *   useValue: (state, slice) => ({ ...state, ...slice })
+ *  }
+ * ]
+ */
+export const RX_ACCUMULATOR_FN = new InjectionToken<AccumulationFn>(
+  'RX_ACCUMULATOR_FN',
+  {
+    providedIn: 'root',
+    factory: () => defaultAccumulator,
+  },
+);
+
+/**
+ * Injection token for the default accumulator function.
+ * @param fn
+ */
+export function withAccumulatorFn(fn: AccumulationFn): RxStateConfigFn {
+  return {
+    kind: RX_STATE_CONFIGS.Accumulator,
+    providers: [{ provide: RX_ACCUMULATOR_FN, useValue: fn }],
+  };
+}
+
+export const RX_STATE_SCHEDULER = new InjectionToken<SchedulerLike | 'sync'>(
+  'RX_STATE_SCHEDULER',
+  {
+    providedIn: 'root',
+    factory: () => queueScheduler,
+  },
+);
+
+/**
+ * Injection token for the default scheduler for rxState.
+ * @param scheduler
+ */
+export function withScheduler(
+  scheduler: SchedulerLike | 'sync',
+): RxStateConfigFn {
+  return {
+    kind: RX_STATE_CONFIGS.Scheduler,
+    providers: [{ provide: RX_STATE_SCHEDULER, useValue: scheduler }],
+  };
+}
+
+/**
+ * Injection token for the default scheduler for rxState.
+ * @param fn
+ */
+export function withSyncScheduler(): RxStateConfigFn {
+  return {
+    kind: RX_STATE_CONFIGS.Scheduler,
+    providers: [{ provide: RX_STATE_SCHEDULER, useValue: 'sync' }],
+  };
+}
```

Comment:
I'm not sure to understand the terminology here, sync is the name for the default scheduler that uses the `queueScheduler` internally, right?

Reply:
no, sync is no scheduler -> sync. it's also described in the docs

Reply:
Ok that's why I was confused, so why the description is "injection token for the default scheduler", seems incorrect.

Reply:
maybe it's worded a little bit weird

Reply:
```suggestion
/**
 * Provide injection token for configuring RxState with the sync scheduler.
 */
export function withSyncScheduler(): RxStateConfigFn {
  return {
    kind: RX_STATE_CONFIGS.Scheduler,
    providers: [{ provide: RX_STATE_SCHEDULER, useValue: 'sync' }],
  };
}
```